src/algorithms/tracking/CMakeLists.txt

```diff
@@ -31,7 +31,7 @@ plugin_glob_all(${PLUGIN_NAME})
 
 # Add libraries
 # (same as target_include_directories but for both plugin and library)
-plugin_link_libraries(${PLUGIN_NAME} Eigen3::Eigen)
+plugin_link_libraries(${PLUGIN_NAME} Eigen3::Eigen ROOT::EG)
```

Comment:
I think ROOT::EG should be added to `plugin_add_cern_root`

https://github.com/eic/EICrecon/blob/main/cmake/jana_plugin.cmake#L219

There are several reasons for this: 

1. ROOG::EG is used for TDatabasePDG, which in turn is used to get charge, and invocations are scattered around many places in code. Moreover, the more algorithms will be added, the more TDatabasePDG is going to be used. So adding ROOG::EG to "global" cmake macro seems reasonable. 

2. You see commented lines there - ROOT::EG is actually a regression to previous problems. And the problems are that some compilations/versions of root have it by default in `${ROOT_LIBRARIES}` while others - don't. And I hoped not to list particular packages.

Reply:
So what I mean here, is if you think reasonable, please switch back to required components in plugin_add_cern_root